### DISCUSSION TITLE
Access from another WSL distribution?

### DISCUSSION THREAD

**Discussion by derkoe** [containers/podman#16130] [Q&A] [answered]
Is it possible to access the podman socket from another WSL distribution?

```
❯ wsl -l -v
  NAME                      STATE           VERSION
* Fedora                    Running         2
  podman-machine-default    Running         2

❯ docker ps
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES

❯ wsl -d Fedora docker ps
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Comment by benoitf** [comment]
Hello @derkoe 

Could you explain it more ?
Because you're saying that you want to access podman socket but you're using docker commands and not podman commands.

**Reply by derkoe** [reply]
The Docker client uses the Podman socket - this works under Windows (because Podman Desktop set things correctly).

How do I have to set `DOCKER_HOST` in another WSL distro to access this Podman socket?

When I use `podman` command in the Fedora distro it runs container in this distro, I want to access the containers in the `podman-machine-default` distro.

**Comment by benoitf** [comment]
It's the "podman machine" code that sets-up the redirect from a local socket to the podman-remote socket running inside WSL

If you need to do that 'by hand' you need to check first, that inside your Fedora WSL, the socket for podman is exposed.
Then, check from within WSL the path of the socket

and then, you would need to do a ssh forward from a local file to the ssh connection within your WSL fedora.

And then you could do `DOCKER_HOST=/path/to/your/local.socket docker ps`


Another option would be do run the podman machine init command with another image (provided by --image-path)

Also I'm not sure why you want to use another WSL distribution here ?
You want to use that WSL distribution in Podman Desktop (like see containers that are in another WSL distribution) or it's to use the CLI

**Reply by derkoe** [reply]
I want to do the same as I do in Docker Desktop - access the same containers from Windows and WSL2. Docker desktop sets up a socket in /mnt/wsl/docker-desktop/shared-sockets that can be accessed from any WSL2 distro.

**Reply by benoitf** [reply]
ok, you want to use podman socket of 'podman-machine-default' from another wsl, not from windows or from the podman WSL (as these 2 already work)

AFAIK, for now, there is no file shared in /mnt/wsl from podman

cc @n1hility @gbraad would it be possible for `podman machine` to setup a file in `/mnt/wsl` as well ?

**Comment by benoitf** [comment]
FYI I moved the question from Podman Desktop to Podman as shared socket should be handled by Podman

**Comment by derkoe** [comment]
I got it working with the following "trick" (that also Docker Desktop uses):

Go into the podman WSL distro (`wsl -d podman-default`) and run:

    $ sudo mount --bind /run/user/1000/podman /mnt/wsl/podman/

This creates a mount point to the Podman socket in the shared WSL2 file system.

After that you can use this from any distribution with:

    $ export DOCKER_HOST=unix:///mnt/wsl/podman/podman.sock
    $ docker version
    Client: Docker Engine - Community
    Version:           20.10.20
    API version:       1.41
    Go version:        go1.18.7
    Git commit:        9fdeb9c
    Built:             Tue Oct 18 18:20:18 2022
    OS/Arch:           linux/amd64
    Context:           default
    Experimental:      true

    Server: linux/amd64/fedora-36
    Podman Engine:
      Version:          4.2.1
      APIVersion:       4.2.1
      Arch:             amd64
      BuildTime:        2022-09-07T21:58:19+02:00
      Experimental:     false
      GitCommit:
      GoVersion:        go1.18.5
      KernelVersion:    5.15.68.1-microsoft-standard-WSL2
      MinAPIVersion:    4.0.0
      Os:               linux
    Conmon:
      Version:          conmon version 2.1.4, commit:
      Package:          conmon-2.1.4-3.fc36.x86_64
    OCI Runtime (crun):
      Version:          crun version 1.6
    commit: 18cf2efbb8feb2b2f20e316520e0fd0b6c41ef4d
    spec: 1.0.0
    +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
      Package:          crun-1.6-2.fc36.x86_64
    Engine:
      Version:          4.2.1
      API version:      1.41 (minimum version 1.24)
      Go version:       go1.18.5
      Git commit:
      Built:            Wed Sep  7 21:58:19 2022
      OS/Arch:          linux/amd64
      Experimental:     false

Would be great if the mount could be setup by the Podman machine automatically.

**Reply by giggio** [reply]
This works. I wish it would be more automatic. They recently merged #19705, which should help.
Rancher Desktop already wires the other WSL distribution as well, via a config. It's easier than setting `DOCKER_HOST` manually.

**Comment by mkane91301** [comment]
Do I need to do something similar to enable Podman Desktop to see my Docker daemon running in Ubuntu so I can use the UI to manage Docker?

**Reply by benoitf** [reply]
@mkane91301 Podman desktop needs to check for this extra location when running on Windows. For now it does only check the socket at the default location provided by docker desktop ( not a shared one)

**Reply by KrunalDatascience** [reply]
In need of a similar usecase. Very new to the world of containerization, podman, linux etc. Any step by step guide to achieve this?

**Comment by n1hility** [comment, marked as answer]
Convert to a feature request: https://github.com/containers/podman/issues/16660

**Reply by gbraad** [reply]
You might also wanna take a look at the duplicate: #15190

**Reply by n1hility** [reply]
ah!, ill close the new one as dup
